app/services/foreman/plugin.rb

```diff
@@ -83,6 +83,12 @@ def find(id)
         registered_plugins[id.to_sym]
       end
 
+      # Evaluates additional setup steps after initial registration
+      # useful when you need to define methods later e.g. from config.to_prepare
+      def setup(id, &block)
```

Comment:
> Another approach would be to store the template helpers as string, which would mean we could perform the constantization later in foreman, without the plugin developer would even need to care and user the setup at all

This is the more usual approach for plugin registration features, as it's also compatible with code reloading.  Typically the extended code (e.g. the renderer) would be calling Foreman::Plugin.all to retrieve the data out of the plugin registations, since they're never reloaded.